src/main/java/net/dv8tion/jda/api/entities/MessageReference.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.api.entities;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.exceptions.InsufficientPermissionException;
+import net.dv8tion.jda.api.exceptions.MissingAccessException;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.internal.JDAImpl;
+import net.dv8tion.jda.internal.requests.CompletedRestAction;
+import net.dv8tion.jda.internal.requests.RestActionImpl;
+import net.dv8tion.jda.internal.requests.Route;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * An object representing a reference in a Discord message.
+ * @see Message#getMessageReference()
+ */
+public class MessageReference
+{
+    private final long messageId;
+    private final long channelId;
+    private final long guildId;
+
+    private final JDA api;
+    private final MessageChannel channel;
+    private final Guild guild;
+    private Message referencedMessage;
+
+    public MessageReference(long messageId, long channelId, long guildId, @Nullable Message referencedMessage, JDA api)
+    {
+        this.messageId = messageId;
+        this.channelId = channelId;
+        this.guildId = guildId;
+        this.referencedMessage = referencedMessage;
+
+        TextChannel tc = api.getTextChannelById(channelId);
+
+        if (tc == null)
+        {
+            this.channel = api.getPrivateChannelById(channelId);
```

Comment:
It might be worth checking if the `referencedMessage` has a channel attached and using that directly if available